include/aspect/particle/integrator/rk_2.h

```diff
@@ -0,0 +1,132 @@
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_integrator_rk_2_h
+#define __aspect__particle_integrator_rk_2_h
+
+#include <aspect/particle/integrator/interface.h>
+
+#include <aspect/simulator_access.h>
+
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Integrator
+    {
+      /**
+       * Runge Kutta second order integrator, where y_{n+1} = y_n + dt*v(0.5*k_1), k_1 = dt*v(y_n).
+       * This scheme requires storing the original location, and the read/write_data functions reflect this.
+       */
+      template <int dim>
+      class RK2Integrator : public Interface<dim>, public SimulatorAccess<dim>
+      {
+        public:
+          RK2Integrator();
+
+          /**
+           * Perform an integration step of moving the particles of one cell
+           * by the specified timestep dt. Implementations of this function
+           * must update the particle location. Between calls to this function
+           * the velocity at the updated particle positions is evaluated and
+           * passed to integrate_step during the next call.
+           *
+           * @param [in] begin_particle An iterator to the first particle to be moved.
+           * @param [in] end_particle An iterator to the last particle to be moved.
+           * @param [in] old_velocities The velocities at t_n, i.e. before the
+           * particle movement.
+           * @param [in] velocities The velocities at t_{n+1}, i.e. after the
+           * particle movement.
+           * @param [in] dt The length of the integration timestep.
+           */
+          virtual
+          void
+          local_integrate_step(const typename std::multimap<LevelInd, Particle<dim> >::iterator &begin_particle,
+                               const typename std::multimap<LevelInd, Particle<dim> >::iterator &end_particle,
+                               const std::vector<Tensor<1,dim> > &old_velocities,
+                               const std::vector<Tensor<1,dim> > &velocities,
+                               const double dt);
+
+          /**
+           * This function is called at the end of every integration step.
+           * In case of multi-step integrators is signals the beginning of a
+           * new integration step.
+           */
+          virtual void advance_step();
+
+          /**
+           * @return This function returns true if another integration step is required
+           * by the integrator. Its default implementation returns false.
+           */
+          virtual bool continue_integration() const;
```

Comment:
Since you have a concrete implementation here, no need to state what the default implementation is -- but maybe say what *this* function does.

The documentation will be formatted slightly oddly because the function has no general documentation, just the `@return` tag.